packages/format-library/src/non-breaking-space/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { insert } from '@wordpress/rich-text';
+import { RichTextShortcut } from '@wordpress/block-editor';
+import { isAppleOS } from '@wordpress/keycodes';
+
+const name = 'core/non-breaking-space';
+const title = __( 'Non breaking space' );
+
+export const nonBreakingSpace = {
+	name,
+	title,
+	tagName: 'span',
+	className: 'nbsp',
+	edit( { value, onChange } ) {
+		function addSpace() {
+			onChange( insert( value, ' ' ) );
+		}
+
+		// Shift + Ctrl + Space shortcut is built in on the Mac.
+		if ( isAppleOS() ) {
+			return;
+		}
+
+		return (
+			<RichTextShortcut
+				type="primaryShift"
```

Comment:
Btw, the original issue had a typo. It should be Shift+Alt+Space. But apparently Alt+Space also works so we could simplify it to just Alt+Space?

Reply:
As shown in [this source](https://defkey.com/what-means/alt-space), `Alt+Space` is often used to access the menu of the program itself.

For example, in Chrome, it would appear as follows (this was also the case in FireFox and Edge):

![chrome](https://user-images.githubusercontent.com/54422211/184315942-e6659c50-85fd-4943-b8cf-d52a0ce0fde9.png)


`Shift+Alt+Space` also showed the same behavior.

If we override this shortcut, we will not be able to access the browser menu unless we remove focus from RichText.
I recommend `Ctrl+Shift+Space`, which is a continuation of the Microsoft Word shortcut.

Reply:
`Shift+Alt+Space` has the same behaviour? I think it's fine to override the Shift combination then.

I think we should stick to an Alt combination because Alt indicates an alternative for the current key/character. I think this is the same for Windows? Alt combinations input different characters, whereas Ctrl is more for commands/controls.

Reply:
> I think we should stick to an Alt combination because Alt indicates an alternative for the current key/character. I think this is the same for Windows? Alt combinations input different characters, whereas Ctrl is more for commands/controls.

Yes, that is correct.
But unfortunately, `Alt+Space` (`Alt+Shift+Space`) didn't respond to triggering `RichtextShortcut`, and the browser menu opens.

From my research [here](https://support.microsoft.com/en-us/windows/keyboard-shortcuts-in-windows-dcc61a57-8ff0-cffe-9796-cb9706c75eec), it appears that `Alt + Space` is not a browser shortcut, but is implemented at the OS level.
> Alt + Spacebar | Open the shortcut menu for the active window.

Then, possible patterns and possibilities are as follows.

- ✅ `Ctrl + Shift + Space`: Confirmed to work correctly
- ✅ `Ctrl + Space`: Confirmed to work correctly
- ❗`Shift + Space`: Switch character width between full-width and half-width. (OS shortcut)
- ❗`Ctrl + Space`: Turn the Chinese input method editor (IME) on or off. (OS shortcut)